File: app/models/schemas/help.py
"""Help system schemas."""
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, Field


# --- HelpDescription schemas ---

class HelpDescriptionResponse(BaseModel):
    id: int
    topic_id: int
    section_number: int
    detailed_text: str
    created_at: datetime

    model_config = {"from_attributes": True}


# --- HelpCategory schemas ---

class HelpCategoryBase(BaseModel):
    name: str = Field(min_length=1, max_length=255)
    display_order: int = 0
    is_active: bool = True


class HelpCategoryCreate(HelpCategoryBase):
    pass


class HelpCategoryUpdate(BaseModel):
    name: Optional[str] = Field(default=None, min_length=1, max_length=255)
    display_order: Optional[int] = None
    is_active: Optional[bool] = None


class HelpCategoryResponse(HelpCategoryBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


# --- HelpTopic schemas ---

class HelpTopicBase(BaseModel):
    category_id: int
    title: str = Field(min_length=1, max_length=500)
    content: str = Field(min_length=1)
    display_order: int = 0
    is_active: bool = True


class HelpTopicCreate(HelpTopicBase):
    pass


class HelpTopicUpdate(BaseModel):
    category_id: Optional[int] = None
    title: Optional[str] = Field(default=None, min_length=1, max_length=500)
    content: Optional[str] = Field(default=None, min_length=1)
    display_order: Optional[int] = None
    is_active: Optional[bool] = None


class HelpTopicResponse(HelpTopicBase):
    id: int
    created_at: datetime
    updated_at: datetime
    descriptions: List[HelpDescriptionResponse] = []

    model_config = {"from_attributes": True}


class HelpTopicListResponse(BaseModel):
    items: List[HelpTopicResponse]
    total: int
    skip: int
    limit: int


class HelpSearchResult(BaseModel):
    id: int
    title: str
    content: str
    category_id: int
    category_name: Optional[str] = None

    model_config = {"from_attributes": True}

File: frontend/src/api/help.ts
import client from './client';

export interface HelpDescription {
  id: number;
  topic_id: number;
  section_number: number;
  detailed_text: string;
  created_at: string;
}

export interface HelpCategory {
  id: number;
  name: string;
  display_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface HelpTopic {
  id: number;
  category_id: number;
  title: string;
  content: string;
  display_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  descriptions: HelpDescription[];
}

export interface HelpTopicListResponse {
  items: HelpTopic[];
  total: number;
  skip: number;
  limit: number;
}

export async function getTopics(skip = 0, limit = 100): Promise<HelpTopicListResponse> {
  const response = await client.get('/help/topics', { params: { skip, limit } });
  return response.data;
}

export async function getTopic(id: number): Promise<HelpTopic> {
  const response = await client.get(`/help/topics/${id}`);
  return response.data;
}

export async function searchTopics(query: string, skip = 0, limit = 100): Promise<HelpTopicListResponse> {
  const response = await client.get('/help/search', { params: { q: query, skip, limit } });
  return response.data;
}

export async function getCategories(): Promise<HelpCategory[]> {
  const response = await client.get('/help/categories');
  return response.data;
}

export async function getTopicsByCategory(categoryId: number, skip = 0, limit = 100): Promise<HelpTopicListResponse> {
  const response = await client.get(`/help/categories/${categoryId}/topics`, { params: { skip, limit } });
  return response.data;
}
